piece-retriever/bin/piece-retriever.js

```diff
@@ -110,6 +114,30 @@ export default {
         { signal: request.signal },
       )
 
+      if (originResponse.status >= 500) {
+        ctx.waitUntil(
+          logRetrievalResult(env, {
+            cacheMiss,
+            responseStatus: originResponse.status,
+            egressBytes: 0,
+            requestCountryCode,
+            timestamp: requestTimestamp,
+            dataSetId,
+          }),
+        )
+        const response = new Response(
+          `Service provider ${serviceProviderId} is unavailable at ${url}`,
```

Comment:
Can we please include the HTTP status code returned by the SP?

The current error message will make it difficult for clients and SPs to understand what exactly happened.